github/pkg/apis/sources/v1alpha1/githubsource_validation.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+func (g *GitHubSource) Validate(ctx context.Context) *apis.FieldError {
+	return g.Spec.Validate(ctx).ViaField("spec")
+}
+
+func (gs *GitHubSourceSpec) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+
+	// TODO: there are more requirements for GitHubSource. Add them here.
```

Comment:
Could you file an issue to actually validate the github source resource and link the issue here?

Reply:
yes sir https://github.com/knative/eventing-contrib/issues/827